### PR TITLE
[flutter_tools] support coverage and machine together

### LINKED ISSUE #30108: The test command doesn't support --machine and coverage together
Running `flutter test` with both machine output and coverage data is not supported now. This means that it's necessary to run tests twice which can take some time. Is it possible to make this command accept both options at the same time?

```
$ flutter test --machine --coverage
Running "flutter packages get" in xxxxx...                     1.0s
The test command doesn't support --machine and coverage together
```

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Create a parent/delegate property of the machine test watcher so that it can still call coverage methods. Prevent extra logging in machine mode.

Fixes https://github.com/flutter/flutter/issues/30108

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/test/coverage_collector.dart [hunk 1]
```diff
@@ -19,17 +18,29 @@ import 'watcher.dart';
...
   bool Function(String) libraryPredicate;
 
   @override
   Future<void> handleFinishedTest(ProcessEvent event) async {
-    globals.printTrace('test ${event.childIndex}: collecting coverage');
+    _logMessage('test ${event.childIndex}: collecting coverage');
     await collectCoverage(event.process, event.observatoryUri);
   }
 
+  void _logMessage(String line, { bool error = false }) {
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/test/coverage_collector.dart]: Yeah, a better way to write the tool logging would have everything be structured data by default, and a to have a FriendlyLogger or similar for non-machine that only displays the raw message

**Comment by NanYeYe** [comment]
support coverage and machine together
------------------------------------------
when can I use this feature in flutter stable, now it only can be use with flutter master

**Comment by jmagman** [comment]
@NanYeYe This PR was merged in v1.15.12 and is available in the current stable.  If you are seeing problems running coverage and machine together, please file a new issue and we'll take a look.

**Pull request merged**